### PR TITLE
Feature/add-ssl

### PR DESCRIPTION
**Pull request by minhle-pham**
Add gnutls and rsyslog-gnutls

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,10 +10,12 @@ EXPOSE 5514/tcp 5514/udp
 RUN yum update -y && \
     yum install -y wget telnet && \
     wget -qO /etc/yum.repos.d/rsyslog.repo http://rpms.adiscon.com/v8-stable/rsyslog.repo && \
-    yum install -y rsyslog
-
+    yum install -y rsyslog && \
```
Comment (marty-macfly) [inline Dockerfile:13]: Do we need to install rsyslog if we are installing rsyslog-gnutls ?

File: Dockerfile [hunk 2]
```diff
@@ -10,10 +10,12 @@ EXPOSE 5514/tcp 5514/udp
...
+    yum install rsyslog-gnutls -y && \
+    yum install gnutls gnutls-utils -y
 RUN rm -f /etc/rsyslog.d/*.conf
 COPY rsyslog.conf /etc/
 COPY conf.d/ /etc/rsyslog.d/
+COPY ssl/ /etc/ssl/
```
Comment (marty-macfly) [inline Dockerfile:19]: Don't want to hardcode certificate in the image they need to be provide in each use case

File: ssl/ca.crt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (marty-macfly) [inline ssl/ca.crt:1]: Need to remove those certificate from a base image, it will be added where it's needed but not in the base image.

**Comment by minhle-pham** [comment]
I was remove ssl directory and clean the code in Dockerfile.

File: Dockerfile [hunk 4]
```diff
@@ -10,12 +10,10 @@ EXPOSE 5514/tcp 5514/udp
 RUN yum update -y && \
     yum install -y wget telnet && \
     wget -qO /etc/yum.repos.d/rsyslog.repo http://rpms.adiscon.com/v8-stable/rsyslog.repo && \
-    yum install -y rsyslog && \
     yum install rsyslog-gnutls -y && \
     yum install gnutls gnutls-utils -y
```
Comment (marty-macfly) [inline Dockerfile:14]: They don't come by dependency when installing rsyslog-gnutls ?

Comment (minhle-pham) [reply Dockerfile:14]: I'm sorry my mistake, rsyslog-gnutls is a plugin for rsyslog and still need rsyslog to install first. I was add rsyslog back in new commit.

**Pull request closed without merging**